src/ICS20Transfer.sol

```diff
@@ -39,10 +41,25 @@ contract ICS20Transfer is IIBCApp, IICS20Transfer, IICS20Errors, Ownable, Reentr
         string memory sender = Strings.toHexString(msg.sender);
         string memory sourcePort = "transfer"; // TODO: Find a way to figure out the source port
         bytes memory packetData;
+
+        // If the denom string resolve properly to an address, that means we are passing a native ERC20 tokens
+        (address _contractAddress, bool _success) = ICS20Lib.hexStringToAddress(msg_.denom);
+        if (!_success) {
+            // If the denom string doesn't resolve properly to an address, it implies that the denom is a sdkCoin denom
+            // and
+            // that we have created its ERC20 representation. So we must retrieve it from the foreign contract address
+            _contractAddress = address(_foreignDenomContracts[msg_.denom]);
+        }
+
+        // Use the _sdkCoinAmount to populate the packetData with a uint256 representation of the uint64 supported
+        // in the cosmos world that consider the proper decimals conversions.
+        (uint64 _sdkCoinAmount,) = SdkCoin._ERC20ToSdkCoin_ConvertAmount(_contractAddress, msg_.amount);
```

Comment:
Each time we discard a return variable, we should explain why on the comment above.

Reply:
Added a comment to that effect